File: Java/src/BinaryTreeLevelOrderTraversalTest.java
import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;
import java.util.Collections;

public class BinaryTreeLevelOrderTraversalTest {

    @Test
    public void testLevelOrder() {
        BinaryTreeLevelOrderTraversal b = new BinaryTreeLevelOrderTraversal();

        Assert.assertTrue(b.levelOrder(null).isEmpty());

        BinaryTreeLevelOrderTraversal.TreeNode node0 = new BinaryTreeLevelOrderTraversal.TreeNode(1);
        Assert.assertEquals(b.levelOrder(node0), Collections.singletonList(Collections.singletonList(1)));

        BinaryTreeLevelOrderTraversal.TreeNode node1 = new BinaryTreeLevelOrderTraversal.TreeNode(3);
        node1.left = new BinaryTreeLevelOrderTraversal.TreeNode(9);
        node1.right = new BinaryTreeLevelOrderTraversal.TreeNode(20);
        node1.right.left = new BinaryTreeLevelOrderTraversal.TreeNode(15);
        node1.right.right = new BinaryTreeLevelOrderTraversal.TreeNode(7);

        Assert.assertEquals(b.levelOrder(node1), Arrays.asList(Collections.singletonList(3), Arrays.asList(9, 20), Arrays.asList(15, 7)));

    }

}

File: Kotlin/src/BinaryTreeLevelOrderTraversalTest.kt
import org.junit.Assert
import org.junit.Test

class BinaryTreeLevelOrderTraversalTest {

    @Test
    fun testLevelOrder() {
        val b = BinaryTreeLevelOrderTraversal()

        Assert.assertTrue(b.levelOrder(null).isEmpty())

        val node0 = BinaryTreeLevelOrderTraversal.TreeNode(1)
        Assert.assertEquals(b.levelOrder(node0), listOf(listOf(1)))

        val node1 = BinaryTreeLevelOrderTraversal.TreeNode(3).apply {
            left = BinaryTreeLevelOrderTraversal.TreeNode(9)
            right = BinaryTreeLevelOrderTraversal.TreeNode(20).apply {
                left = BinaryTreeLevelOrderTraversal.TreeNode(15)
                right = BinaryTreeLevelOrderTraversal.TreeNode(7)
            }
        }

        Assert.assertEquals(b.levelOrder(node1), listOf(listOf(3), listOf(9, 20), listOf(15, 7)))
    }

}